### ISSUE TITLE
Android: The `onProgress` event progressively fires more frequently each time the `paused` prop is toggled

### ISSUE THREAD

**Issue by johnbillion** [TheWidlarzGroup/react-native-video#2934]
# Bug

Each time the value of the `paused` prop is toggled, for example via a Play/Pause button, the frequency with which the `onProgress` callback fires will increase, but only on Android. This can be seen by logging the value of `currentTime` to the console from the `onProgress` callback.

## Platform

Which player are you experiencing the problem on:

* Android

## Environment info

React native info output:

```bash
System:
    OS: macOS 12.3
    CPU: (10) arm64 Apple M1 Pro
    Memory: 1.18 GB / 32.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.21.1 - ~/.nvm/versions/node/v14.21.1/bin/node
    Yarn: Not Found
    npm: 6.14.17 - ~/.nvm/versions/node/v14.21.1/bin/npm
    Watchman: 2022.10.24.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK: Not Found
  IDEs:
    Android Studio: 2021.3 AI-213.7172.25.2113.9123335
    Xcode: 13.4.1/13F100 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.17 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.1.0 => 18.1.0
    react-native: ^0.70.6 => 0.70.6
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

Library version: `6.0.0-alpha.3`

## Steps To Reproduce

See below for a minimum sample component that reproduces the problem.

1. Add an `onProgress` callback
2. Implement a means of toggling the `paused` prop, for example a play/pause button
3. Observe on Android that each time the `paused` prop is toggled, the `onProgress` callback fires twice as frequently than it should
4. Rinse and repeat

## Expected behaviour

1. The frequency with which the `onProgress` event fires should remain stable

## Reproducible sample code

```javascript
import { React, useState } from 'react';
import { SafeAreaView, Text, TouchableOpacity } from 'react-native';
import Video from 'react-native-video';

const App = () => {
	const [isPaused, setPaused] = useState(false);

	return (
		<SafeAreaView>
			<Video
				source={{
					uri: 'https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4',
				}}
				style={{
					width: 320,
					height: 240,
				}}
				paused={isPaused}
				onProgress={(data) => {
					console.log(data.currentTime);
				}}
				progressUpdateInterval={1000}
			/>
			<TouchableOpacity
				onPress={() => {
					setPaused( !isPaused );
					console.log(isPaused ? 'Resumed' : 'Paused')
				}}
			>
				<Text>
					{isPaused ? 'Play' : 'Pause'}
				</Text>
			</TouchableOpacity>
		</SafeAreaView>
	);
};

export default App;
```

## Video sample

I don't have a video sample, but see below for the console log that shows the frequency of the the `onProgress` event increasing. The console should log once per second, and does so until the video is paused and resume, upon which it logs 3X too often.

```
 LOG  Running "nativevideo" with {"rootTag":1}
 LOG  0
 LOG  0.875
 LOG  1.878
 LOG  2.876
 LOG  3.879
 LOG  Paused
 LOG  4.478
 LOG  Resumed
 LOG  4.51
 LOG  4.699
 LOG  5.298
 LOG  5.539
 LOG  5.707
 LOG  6.303
 LOG  6.544
 LOG  6.703
 LOG  7.309
 LOG  7.341
 LOG  7.507
 LOG  7.749
 LOG  7.909
 LOG  8.515
 LOG  8.74
 LOG  8.929
 LOG  9.505
 LOG  9.747
 LOG  9.928
 LOG  10.511
 LOG  10.747
 LOG  10.927
```

**Comment by cross19xx** [comment]
Could you provide a code repro. I have a feeling other components may be causing rerenders

**Comment by freeboub** [comment]
@cross19xx with the updated sample in this PR : https://github.com/react-native-video/react-native-video/pull/2960
It can be reproduced.
I have a quick look into the code, some messages are not cleared (SHOW_PROGRESS) . It causes the issue

**Comment by freeboub** [comment]
Fix available here: https://github.com/react-native-video/react-native-video/pull/2972

**Comment by freeboub** [comment]
fixed in 6.0.0-alpha.5

**Comment by vishalhyperspace** [comment]
I have still same issue. Please help me to fixed. If i remove onProgress event it will work fine.
